cloudflare.go

```diff
@@ -212,7 +212,12 @@ func (api *API) makeRequestWithAuthTypeAndHeaders(ctx context.Context, method, u
 			}
 			// useful to do some simple logging here, maybe introduce levels later
 			api.logger.Printf("Sleeping %s before retry attempt number %d for request %s %s", sleepDuration.String(), i, method, uri)
-			time.Sleep(sleepDuration)
+
+			select {
+			case <-time.After(sleepDuration):
+			case <-ctx.Done():
+				return nil, errors.Wrap(ctx.Err(), "context deadline exceeded during retries")
```

Comment:
@favonia `"context deadline exceeded during retries"` is a somewhat misleading message: the `ctx` might be `Done()` due to a cancel of a parent `context.WithCancel`. I suggest rephrasing ( or just not wrapping at all ).

Reply:
Ah, good point. Any suggestion? I think it's still better to wrap external errors, but I am not a maintainer here, so I don't have a strong opinion.

Reply:
Alright, I pushed something which hopefully addressed the issue.

Reply:
@favonia The wording is great thank you!